src/engines/contract_engine.rs

```diff
@@ -171,20 +171,48 @@ pub async fn dry_run_call(
 mod tests {
 	use super::*;
 	use std::fs;
+	use std::path::PathBuf;
+
+	fn setup_test_environment() -> anyhow::Result<tempfile::TempDir, anyhow::Error> {
```

Comment:
`use anyhow::Result`, shouldnt then need to specify the error type and can probaby standardise on just Result<T> for all test return types

Reply:
Updated.
```rust
use anyhow::{Result, Error};
```
I had to include `Error` otherwise it was erroring out with the following:
```
error[E0412]: cannot find type `Error` in this scope
   --> src/engines/contract_engine.rs:176:59
    |
176 |     fn setup_test_environment() -> Result<tempfile::TempDir, Error> {
    |                                                              ^^^^^ not found in this scope
    |
help: consider importing one of these items
    |
172 +     use anyhow::Error;
    |
172 +     use askama::Error;
    |
172 +     use clap::Error;
    |
172 +     use clap::error::Error;
    |
      and 18 other candidates
```